### PR TITLE
[software-bridges 4/x] Controller part

### PR DESCRIPTION
**Pull request by ykulazhenkov**
This PR contains controller's part of the software bridge management feature.

Context is [Software bridge management feature](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/doc/design/software-bridge-management.md)

### CODE REVIEW

**Review by adrianchiris — APPROVED** [review]
LGTM

**Comment by ykulazhenkov** [comment]
@SchSeba I resolved merge conflicts in the PR, please take a look

File: api/v1/helper.go [hunk 1]
```diff
@@ -438,6 +439,61 @@ func (p *SriovNetworkNodePolicy) Apply(state *SriovNetworkNodeState, equalPriori
...
+			return fmt.Errorf("linkType must be eth or ETH to use software bridge management")
+		}
+		if p.Spec.ExternallyManaged {
+			return fmt.Errorf("software bridge management can't be used when link is externally managed")
+		}
+	}
+	for _, iface := range state.Status.Interfaces {
+		if p.Spec.NicSelector.Selected(&iface) {
+			if p.Spec.Bridge.OVS == nil {
+				state.Spec.Bridges.OVS = slices.DeleteFunc(state.Spec.Bridges.OVS, func(br OVSConfigExt) bool {
```
Comment (SchSeba) [inline api/v1/helper.go:463]: took me 5 min to understand this can you please add a comment what it does :P

File: api/v1/helper_test.go [hunk 2]
```diff
@@ -1122,3 +1122,415 @@ func TestNeedToUpdateSriov(t *testing.T) {
...
+					NumVfs:            2,
+					Priority:          99,
+					EswitchMode:       "switchdev",
+					ExternallyManaged: true,
+					ResourceName:      "p1res",
+					Bridge:            v1.Bridge{OVS: &v1.OVSConfig{}},
+				},
+			},
+			expectedBridges: v1.Bridges{},
+			expectedErr:     true,
```
Comment (SchSeba) [inline api/v1/helper_test.go:1231]: is externally manage not support at all or right now and will be in the future?

File: pkg/webhook/validate.go [hunk 3]
```diff
@@ -213,6 +213,15 @@ func staticValidateSriovNetworkNodePolicy(cr *sriovnetworkv1.SriovNetworkNodePol
 	if cr.Spec.DeviceType == consts.DeviceTypeVfioPci && cr.Spec.IsRdma {
 		return false, fmt.Errorf("'deviceType: vfio-pci' conflicts with 'isRdma: true'; Set 'deviceType' to (string)'netdevice' Or Set 'isRdma' to (bool)'false'")
 	}
+	if strings.EqualFold(cr.Spec.LinkType, consts.LinkTypeIB) && !cr.Spec.IsRdma {
```
Comment (SchSeba) [inline pkg/webhook/validate.go:216]: I think this one was removed  no?

https://github.com/k8snetworkplumbingwg/sriov-network-operator/commit/78bde32285684f1300d2c689b974edc9421bbe75

**Review by SchSeba — COMMENTED** [review]
small comments and we can merge this one

File: api/v1/helper_test.go (hunk 2 above)
Comment (ykulazhenkov) [reply api/v1/helper_test.go:1231]: If the device is externally managed we are not managing bridges for it, but It is possible to create OVSNetwork for Externally managed PF to announce "manually" created resources. For externally managed PFs we are not changing PF-level settings. For me "software bridge config for PF" option is a PF-level settings and I think we should not change it externally managed for PFs.

File: pkg/webhook/validate.go (hunk 3 above)
Comment (ykulazhenkov) [reply pkg/webhook/validate.go]: Thx! Good catch! done

File: api/v1/helper.go (hunk 1 above)
Comment (ykulazhenkov) [reply api/v1/helper.go:466]: I added a comment to clarify this logic.

**Comment by ykulazhenkov** [comment]
@SchSeba Thx for the review. I addressed your comments, please take another look.

**Review by SchSeba — APPROVED** [review]
Great work!

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**